genologics/lims.py

```diff
@@ -379,21 +380,24 @@ def _get_instances(self, klass, params=dict()):
             root = self.get(node.attrib['uri'], params=params)
         return result
 
-    def get_batch(self, instances):
-        "Get the content of a set of instances using the efficient batch call."
+    def get_batch(self, instances, force=False):
+        """Get the content of a set of instances using the efficient batch call."""
         if not instances:
             return []
-        klass = instances[0].__class__
+
         root = ElementTree.Element(nsmap('ri:links'))
+        klass = None
         result = []
         needs_request=False
         for instance in instances:
-            try:
-                result.append(self.cache[instance.uri])
-            except:
-                needs_request=True
+            if not klass:
```

Comment:
Yeah, I think there is something fishy with the batch calls at the moment, which is why they are off by default in our repo.
I didn't dig that much into it, though, so it might be something obvious that I overlooked.